src/DeviceDataStreamValues.tsx

```diff
@@ -0,0 +1,232 @@
+  if (data.length === 0) {
+    return <p>Maybe device published data but not in the selected timeframe.</p>;
+  }
+
+  return (
+    <>
+      <Table responsive>
+        <thead>
+          <tr>
+            <th>Path</th>
+            <th>Value</th>
+            <th>Timestamp</th>
+          </tr>
+        </thead>
+        <tbody>
+          {data.map((tree) => (
+            <tr key={tree.endpoint}>
+              <td>{tree.endpoint}</td>
+              <td>{formatAstarteData(tree)}</td>
+              <td>{new Date(tree.timestamp).toLocaleString()}</td>
+            </tr>
+          ))}
+        </tbody>
+      </Table>
+    </>
+  );
+};
+
+interface ObjectDatastreamTableProps {
+  data: AstarteDatastreamObjectData[];
+}
+
+const ObjectDatastreamTable = ({ data }: ObjectDatastreamTableProps): React.ReactElement => {
+  if (data.length === 0) {
+    return <p>No data sent by the device.</p>;
+  }
+  const orderedData = _.orderBy(data, 'timestamp', 'desc');
+
+  const objectProperties = Object.keys(orderedData[0].value);
+
+  return (
+    <>
+      <Table responsive>
+        <thead>
+          <tr>
+            {objectProperties.map((property) => (
+              <th key={property}>{property}</th>
+            ))}
+            <th>Timestamp</th>
+          </tr>
+        </thead>
+        <tbody>
+          {orderedData.map((data) => (
+            <tr key={data.timestamp}>
+              {objectProperties.map((property) => (
+                <td key={property}>{formatAstarteData(data.value[property])}</td>
+              ))}
+              <td>{new Date(data.timestamp).toLocaleString()}</td>
+            </tr>
+          ))}
+        </tbody>
+      </Table>
+    </>
+  );
+};
+
+interface InterfaceDataProps {
+  interfaceData: AstarteDatastreamIndividualData[] | AstarteDatastreamObjectData[];
+  aggregation: AstarteInterface['aggregation'];
+}
+
+const InterfaceData = ({ interfaceData, aggregation }: InterfaceDataProps): React.ReactElement => {
+  if (aggregation === 'individual') {
+    return <IndividualDatastreamTable data={interfaceData as AstarteDatastreamIndividualData[]} />;
+  }
+  return <ObjectDatastreamTable data={interfaceData as AstarteDatastreamObjectData[]} />;
+};
+
+export default (): React.ReactElement => {
+  const { deviceId = '', interfaceName = '' } = useParams();
+  const astarte = useAstarte();
+  const [requestingData, setRequestingData] = useState(false);
+  const [fetchedData, setFetchedData] = useState<AstarteInterfaceValues>();
+  const [formAlerts, formAlertsController] = useAlerts();
+  const [interfaceData, setInterfaceData] = useState<AstarteInterface>();
+
+  const deviceDataFetcher = useFetch(() =>
+    astarte.client.getDeviceData({
+      deviceId,
+      interfaceName,
+    }),
+  );
+
+  const interfaceMajor = astarte.client
+    .getInterfaceMajors(interfaceName)
+    .then((interfaceMajors) => {
+      const latestMajor = Math.max(...interfaceMajors);
+      return latestMajor;
```

Comment:
We shouldn't get the latest major. We should get the same major version used by the device.
This can also be a path param set by the previous page